jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -89,10 +97,23 @@ object XGBoost extends Serializable {
     } else {
       throw new IllegalArgumentException(
         s"Encountered a partition with $nUndefined NaN base margin values. " +
-            "If you want to specify base margin, ensure all values are non-NaN.")
+          s"If you want to specify base margin, ensure all values are non-NaN.")
     }
   }
 
+  /**
+   * Create an RDD which is responsible for actually executing the training stage. Each RDD
+   * partition is converted to an XGBoost `DMatrix`, and a gradient boosted decision tree model
+   * is trained by calling into the native XGBoost code through the Java Native Interface.
+   *
+   * Each RDD partition is essentially a long-running task that repeatedly performs boosting
+   * iterations by calling into the native XGBoost library and communicates split statistics
+   * through the [[Rabit]] all-reduce framework instead of performing aggregations directly
+   * within Spark. Training is executed in a single Spark job, and can be triggered by invoking
+   * some action on the returned RDD - e.g. a call to `foreachPartition`.
+   *
+   * Refer to `trainWithRDD` for parameter descriptions.
```

Comment:
I removed the duplicated parameter descriptions from the private methods. I can even remove this "Refer to" line if that's desirable.